apiserver/application/application.go

```diff
@@ -905,7 +973,40 @@ func (api *API) processSameControllerRemoteApplication(url jujucrossmodel.Applic
 	if appName == "" {
 		appName = url.ApplicationName
 	}
-	return api.saveRemoteApplication(sourceModelTag, appName, url.ApplicationName, url.String(), endpoints)
+	remoteApp, err := api.saveRemoteApplication(sourceModelTag, appName, url.ApplicationName, url.String(), endpoints)
+	return remoteApp, sourceModelTag, err
+}
+
+// sameControllerOfferedApplication looks the specified model on the same controller
+// and returns the specified application and a reference to its state.State.
+func (api *API) sameControllerOfferedApplication(sourceModelTag names.ModelTag, applicationName string) (
+	_ *state.Application,
+	closer io.Closer,
+	err error,
+) {
+	defer func() {
+		if err != nil && closer != nil {
+			closer.Close()
+		}
+	}()
+
+	ok, err := api.authorizer.HasPermission(permission.WriteAccess, sourceModelTag)
```

Comment:
you have to have write access to the offered model if you just want to consume from it?